### PR TITLE
docs(config validation): rewrite

### PR DESCRIPTION
**Pull request by HonkingGoose**
@coderabbitai: ignore

## Changes

- Rewrite config validation docs

## Context

I think using lists and more headings is good.

I'm not closing any issue with this PR.

Follow up after PR:

- #21969

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/config-validation.md [hunk 1]
```diff
@@ -18,5 +25,18 @@ INFO: Validating renovate.json
...
-Please check out the [`renovatebot/pre-commit-hooks` repository](https://github.com/renovatebot/pre-commit-hooks) for more information.
+### Pass files to check as CLI arguments
+
+Or you can pass the files to check to the `renovate-config-validator` program as CLI arguments.
+For example:
+
+```console
+$ npm install --global renovate
+added 750 packages, and audited 751 packages in 51s
+$ renovate-config-validator ///// insert CLI arguments here
```
[the remark this answers is not in the document]
Comment (fgreinacher) [reply docs/usage/config-validation.md:36]: @HonkingGoose I only needed it when checking config files with non-default names, e.g. from preset repositories.

[the remark this answers is not in the document]
Comment (HonkingGoose) [reply docs/usage/config-validation.md:37]: Does anyone have an example of how to use that CLI flag? I don't even know what the flag is called. 😄 

I'm looking for something like:

```console
$ command --cli-flag filename, second-filename
$ output for first file check
$ output for second file check
```

[the remark this answers is not in the document]
Comment (fgreinacher) [reply docs/usage/config-validation.md:37]: You don't need a flag, just pass the files that should be checked as arguments:

```terminal
$ renovate-config-validator first_config.json second_config.json
```

[the remark this answers is not in the document]
Comment (HonkingGoose) [reply docs/usage/config-validation.md:37]:
```suggestion
$ renovate-config-validator first_config.json second_config.json
```

File: docs/usage/config-validation.md [hunk 2]
```diff
@@ -5,18 +5,42 @@ description: How to validate Renovate's configuration.
...
+For example:
+
+```console
+$ npm install --global renovate
+added 750 packages, and audited 751 packages in 51s
+$ renovate-config-validator first_config.json second_config.json
+ INFO: Validating first_config_.json
+ INFO: Validating second_config.json
+ INFO: Config validated successfully
+```
```
Comment (HonkingGoose) [inline docs/usage/config-validation.md:41]: When I validate multiple files with the program, it's not clear which file is good and which is bad. 

I was expecting to see output like this:

```console
$ renovate-config-validator good_config.json bad_config.json
 INFO: Validating good_config_.json
 INFO: good_config.json validated successfully
 INFO: Validating bad_config.json
 INFO: bad_config.json has errors
 INFO: finished the validation run
```

I think we should:

- Always print the filename next to the validation result
- As the final step, print a "finished the validation run" message so you know the program exited cleanly

Do you want an issue from me for this?

Comment (HonkingGoose) [reply docs/usage/config-validation.md:46]: @rarkins can you decide if you want an issue (or new discussion) for my idea to list good and bad configs, so it's clear which file passes/fails validation? Or is the expected use-case that users only check one file at a time?

Comment (rarkins) [reply docs/usage/config-validation.md:46]: I don't think it's intended to validate more than one at a time

Comment (HonkingGoose) [reply docs/usage/config-validation.md:46]: @rarkins Choose your own adventure:

1. @HonkingGoose creates issue to make config validation work with multiple files (see example output above), and updates the docs for now to say "check one file at a time".
2. @HonkingGoose only updates the docs to say "check one file at a time".

Comment (fgreinacher) [reply docs/usage/config-validation.md:46]: We do use this to validate multiple files, in a repository that hosts config presets. I think it's a valuable use case.

Comment (HonkingGoose) [reply docs/usage/config-validation.md]: I tried validating multiple files with the program, and it _seems_ to work, but the output is very confusing, quoting myself:

> When I validate multiple files with the program, it's not clear which file is good and which is bad. 
> 
> I was expecting to see output like this:
> 
> ```console
> $ renovate-config-validator good_config.json bad_config.json
>  INFO: Validating good_config_.json
>  INFO: good_config.json validated successfully
>  INFO: Validating bad_config.json
>  INFO: bad_config.json has errors
>  INFO: finished the validation run
> ```
> 
> I think we should:
> 
> - Always print the filename next to the validation result
> - As the final step, print a "finished the validation run" message so you know the program exited cleanly

But the maintainer wanted me to edit the docs. And I did. 🙃

Comment (rarkins) [reply docs/usage/config-validation.md]: So what's the output today?

Comment (HonkingGoose) [reply docs/usage/config-validation.md]: ## Checking two files by name

```bash
$ npm install --global renovate
$ $ renovate-config-validator good-renovate.json bad-renovate.json 
 INFO: Validating good-renovate.json
 INFO: Validating bad-renovate.json
ERROR: Found errors in configuration
       "file": "bad-renovate.json",
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Invalid configuration option: packageRules[0].descriptio"
         }
       ]
```

### `good-renovate.json`

```json
{
  "extends": ["config:best-practices"],
  "packageRules": [
    {
      "description": "Wait 3 days before creating a npm update PR",
      "matchDatasources": ["npm"],
      "minimumReleaseAge": "3 days"
    }
  ]
}
```

### `bad-renovate.json`

```json
{
    "extends": ["config:best-practices"],
    "packageRules": [
      {
        "descriptio": "Wait 3 days before creating a npm update PR",
        "matchDatasources": ["npm"],
        "minimumReleaseAge": "3 days"
      }
    ]
  }
```

## Default run with good `renovate.json` in repo root

```bash
$ renovate-config-validator
 INFO: Validating renovate.json
 INFO: Config validated successfully
```

## Default run with bad `renovate.json` in repo root

```bash
$ renovate-config-validator
 INFO: Validating renovate.json
ERROR: Found errors in configuration
       "file": "renovate.json",
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "Invalid configuration option: packageRules[0].descriptio"
         }
       ]
```

Comment (HonkingGoose) [reply docs/usage/config-validation.md]: @rarkins can you take a look at the comment above, and tell me if you want an issue/discussion for the Renovate config validator output?

Comment (rarkins) [reply docs/usage/config-validation.md]: @HonkingGoose is the above examples a mockup of possible future behavior, or showing current?

You wrote earlier:
> it's not clear which file is good and which is bad

But to me your recent examples seem pretty clear

Comment (HonkingGoose) [reply docs/usage/config-validation.md]: That's current behavior.

Now I'm digging deeper into the error message itself, it does say: `"file": "bad-renovate.json",` so it points to the bad file. Let's keep things as they are then. 😉 Sorry for the confusion!

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**